modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/APIResourceManagementClient.java

```diff
@@ -0,0 +1,296 @@
+    private final String apiResourceManagementBasePath;
+
+    public APIResourceManagementClient(String serverUrl, Tenant tenantInfo) {
+
+        this.serverUrl = serverUrl;
+        this.tenantDomain = tenantInfo.getContextUser().getUserDomain();
+        this.username = tenantInfo.getContextUser().getUserName();
+        this.password = tenantInfo.getContextUser().getPassword();
+
+        apiResourceManagementBasePath = getAPIResourceMgtPath(serverUrl, tenantDomain);
+    }
+
+    private String getAPIResourceMgtPath(String serverUrl, String tenantDomain) {
+
+        if (tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+            return serverUrl + API_SERVER_PATH + PATH_SEPARATOR + API_RESOURCE_MANAGEMENT_PATH;
+        }
+        return serverUrl + TENANT_PATH + tenantDomain + PATH_SEPARATOR + API_SERVER_PATH + PATH_SEPARATOR +
+                API_RESOURCE_MANAGEMENT_PATH;
+    }
+
+    /**
+     * Create an API Resource.
+     *
+     * @param apiResourceCreationModel API Resource creation model.
+     * @return API Resource ID.
+     * @throws Exception If an error occurred while creating the API Resource.
+     */
+    public String createAPIResource(APIResourceCreationModel apiResourceCreationModel) throws Exception {
+
+        String jsonRequestBody = toJSONString(apiResourceCreationModel);
+
+        try (CloseableHttpResponse response = getResponseOfHttpPost(apiResourceManagementBasePath, jsonRequestBody,
+                getHeadersWithBasicAuth())) {
+            String[] locationElements = response.getHeaders(LOCATION_HEADER)[0].toString().split(PATH_SEPARATOR);
+            return locationElements[locationElements.length - 1];
+        }
+    }
+
+    /**
+     * Get API resources by filtering.
+     *
+     * @param apiResourceFilter API resource filter.
+     * @return List of API resources.
+     * @throws IOException Error when getting the filtered API resource.
+     */
+    public List<APIResourceListItem> getAPIResourcesWithFiltering(String apiResourceFilter) throws IOException {
+
+        String endPointUrl = apiResourceManagementBasePath;
+        if (StringUtils.isNotBlank(apiResourceFilter)) {
+            endPointUrl = apiResourceManagementBasePath + "?filter=" + apiResourceFilter;
+        }
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl, getHeadersWithBasicAuth())) {
+            String responseBody = EntityUtils.toString(response.getEntity());
+            ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+            APIResourceListResponse apiResourceListResponse =
+                    jsonWriter.readValue(responseBody, APIResourceListResponse.class);
+            return apiResourceListResponse.getApiResources();
+        }
+    }
+
+    /**
+     * Get API resources by filtering from a sub organization.
+     *
+     * @param apiResourceFilter API resource filter.
+     * @param switchedM2MToken Switched M2M token.
```

Comment:
Let's fix formatting here

Reply:
fixed in [fea9dce](https://github.com/wso2/product-is/pull/23306/commits/fea9dce37c558217627f44ca67866f3ed22d81f9)